src/main/java/com/microsoft/azure/vmagent/AzureVMMaintainPoolTask.java

```diff
@@ -54,7 +55,17 @@ public void maintain(AzureVMCloud cloud, AzureVMAgentTemplate template) {
     }
 
     public void provisionNodes(AzureVMCloud cloud, AzureVMAgentTemplate template, int newAgents) {
-        cloud.doProvision(newAgents, new ArrayList<NodeProvisioner.PlannedNode>(), template, true);
+        if (cloud.getConfigurationStatus().equals(Constants.UNVERIFIED)
+                || template.getTemplateConfigurationStatus().equals(Constants.UNVERIFIED)) {
+            AzureVMCloudVerificationTask.verify(cloud.getCloudName(), template.getTemplateName());
```

Comment:
Unless I missed something, if the pool size of a template is 0 (scale to 0), I didn't see anywhere where a failed verification is reset to unverified.

Reply:
Failed won't reset to unverified automatically unless user save the configurations again. Do you think it's too strict? But according to my experience, verification rarely failed by accident issues. So it's make sense to fix the configuration issues when verification failed.
Thus, whether it's a bad template or not, we will verify it once in the most time. It reduce most of useless Azure API call in this PR.

Reply:
@zackliu Hmm...I have seen that happen before (e.g. Azure rate limits), though not often.  If an allocation fails, then reset happens automatically right?

This may be okay.